### ISSUE TITLE
"no such file or directory"

### ISSUE THREAD

**Issue by SturmB** [zsh-users/zsh-syntax-highlighting#603]
I am unable to get this to work on one of my two Ubuntu boxes. I followed the exact same procedure for both, using the instructions for `oh-my-zsh`, which is installed on both. However, on one of them, after updating `~/.zshrc` and sourcing it, I am greeted with this message:
```
 sturm@SkyUbuntu  ~  source ~/.zshrc
/home/sturm/.oh-my-zsh/custom/plugins/zsh-syntax-highlighting/zsh-syntax-highlighting.plugin.zsh:source:2: no such file or directory: /home/sturm/.oh-my-zsh/custom/plugins/zsh-syntax-highlighting/zsh-syntax-highlighting.zsh^M
 sturm@SkyUbuntu  ~ 
```
To me, it looks like the DOS line endings is causing issues. I've tried converting the `zsh-syntax-highlighting.plugin.zsh` file to use Mac line endings, but the issue persists.

What's most frustrating is that this did _not_ occur on our other Ubuntu box, which has the exact same versions of everything installed. (They are practically identical.) I even ran the `apt` update/upgrade process on both of them before installing this plugin to make sure the playing field was level. What gives?

**Comment by danielshahaf** [comment]
DOS/Mac/Unix line endings are CRLF/CR/LF respectively, so switching to Mac line endings won't help. Try switching it to "Unix" line endings and confirm with `</home/sturm/.oh-my-zsh/custom/plugins/zsh-syntax-highlighting/zsh-syntax-highlighting.plugin.zsh xxd -p -c1 | grep '0[da]' | sort | uniq`.

As to the cause, it's a wild guess but try diffing the outputs of `git config --get-regexp '^'`.

**Comment by SturmB** [comment]
I'm sorry; I'm still very much the newbie, so I do not know how to diff, I'm afraid. The `git config --get-regexp '^'` command in the `/home/sturm/.oh-my-zsh/custom/plugins/zsh-syntax-highlighting` directory doesn't really tell me anything of value:
```
core.autocrlf true
user.email chris@chrismcgee.info
user.name Chris McGee
push.default simple
core.repositoryformatversion 0
core.filemode true
core.bare false
core.logallrefupdates true
remote.origin.url https://github.com/zsh-users/zsh-syntax-highlighting.git
remote.origin.fetch +refs/heads/*:refs/remotes/origin/*
branch.master.remote origin
branch.master.merge refs/heads/master
(END)
```

I did, however, try your first suggestion (without the `<`, since that didn't do anything at all, from what I could tell). The result:
```
 sturm@SkyUbuntu  ~  /home/sturm/.oh-my-zsh/custom/plugins/zsh-syntax-highlighting/zsh-syntax-highlighting.plugin.zsh xxd -p -c1 | grep '0[da]' | sort | uniq
zsh: permission denied: /home/sturm/.oh-my-zsh/custom/plugins/zsh-syntax-highlighting/zsh-syntax-highlighting.plugin.zsh
 sturm@SkyUbuntu  ~ 
```
Don't know why permission is denied, though, as it appears to have plenty of write access:
```
 sturm@SkyUbuntu  ~  l ~/.oh-my-zsh/custom/plugins/zsh-syntax-highlighting
total 116K
drwxrwxr-x 7 sturm sturm 4.0K Feb  7 10:28 .
drwxr-xr-x 4 sturm sturm 4.0K Feb  7 09:59 ..
-rw-rw-r-- 1 sturm sturm  15K Feb  7 09:59 changelog.md
-rw-rw-r-- 1 sturm sturm 1.6K Feb  7 09:59 COPYING.md
drwxrwxr-x 3 sturm sturm 4.0K Feb  7 09:59 docs
drwxrwxr-x 8 sturm sturm 4.0K Feb  7 11:18 .git
-rw-rw-r-- 1 sturm sturm   29 Feb  7 09:59 .gitattributes
-rw-rw-r-- 1 sturm sturm   27 Feb  7 09:59 .gitignore
-rw-rw-r-- 1 sturm sturm 2.3K Feb  7 09:59 HACKING.md
drwxrwxr-x 9 sturm sturm 4.0K Feb  7 09:59 highlighters
drwxrwxr-x 2 sturm sturm 4.0K Feb  7 09:59 images
-rw-rw-r-- 1 sturm sturm 4.9K Feb  7 09:59 INSTALL.md
-rw-rw-r-- 1 sturm sturm 2.0K Feb  7 09:59 Makefile
-rw-rw-r-- 1 sturm sturm 3.0K Feb  7 09:59 README.md
-rw-rw-r-- 1 sturm sturm  806 Feb  7 09:59 release.md
-rw-rw-r-- 1 sturm sturm   13 Feb  7 09:59 .revision-hash
drwxrwxr-x 2 sturm sturm 4.0K Feb  7 09:59 tests
-rw-rw-r-- 1 sturm sturm 1.1K Feb  7 09:59 .travis.yml
-rw-rw-r-- 1 sturm sturm   11 Feb  7 09:59 .version
-rw-rw-r-- 1 sturm sturm   59 Feb  7 10:28 zsh-syntax-highlighting.plugin.zsh
-rw-rw-r-- 1 sturm sturm  18K Feb  7 10:28 zsh-syntax-highlighting.zsh
 sturm@SkyUbuntu  ~ 
```

**Comment by danielshahaf** [comment]
> I'm sorry; I'm still very much the newbie, so I do not know how to diff, I'm afraid. The `git config --get-regexp '^'` command in the `/home/sturm/.oh-my-zsh/custom/plugins/zsh-syntax-highlighting` directory doesn't really tell me anything of value:

"To diff" means "to compare". I meant you to run that command in both environments (the working one and the non-working one) and compare the outputs.

> I did, however, try your first suggestion (without the `<`, since that didn't do anything at all, from what I could tell).

Please try it again _with_ the `<`.  Without that, it will simply error out (as you discovered) since `…/zsh-syntax-highlighting.plugin.zsh` is not a valid command.  The syntax `<foo bar` is equivalent to `bar < foo`.

tl;dr You need to get the .plugin.zsh files to use Unix-style (LF, 0x0A) line endings.

**Comment by danielshahaf** [comment]
The expected output of the command is `0a` and nothing else.

**Comment by SturmB** [comment]
In that case, I see both `0a` and `0d`:
```
 sturm@SkyUbuntu  ~  </home/sturm/.oh-my-zsh/custom/plugins/zsh-syntax-highlighting/zsh-syntax-highlighting.plugin.zsh xxd -p -c1 | grep '0[da]' | sort | uniq
0a
0d
 sturm@SkyUbuntu  ~ 
```
I assume those are the ANSI/ASCII/Unicode/Whatever representations for CR and LF. On our other Ubuntu box, the same command returns just `0a`.

**Comment by danielshahaf** [comment]
> In that case, I see both `0a` and `0d`:
> ```
>  sturm@SkyUbuntu  ~  </home/sturm/.oh-my-zsh/custom/plugins/zsh-
> syntax-highlighting/zsh-syntax-highlighting.plugin.zsh xxd -p -c1 | grep 
> '0[da]' | sort | uniq
> 0a
> 0d
>  sturm@SkyUbuntu  ~ 
> ```
> I assume those are the ANSI/ASCII/Unicode/Whatever representations for 
> CR and LF.

ASCII and UTF-8.

> On our other Ubuntu box, the same command returns just `0a`.

Yes, that's what it should print.  You need to check the setup on the other non-working box to see why it injects a 0d in there.  It could be because of the core.autocrlf in the git setup as I mentioned.

At this point, this appears not to be a z-sy-h bug at all.

**Comment by SturmB** [comment]
That was it, indeed. I set `core.autocrlf` to `input` instead of `true`, re-did the `git clone` of the repo, and now `0a` is the only response to your debugging line. Now the plugin is loading up correctly. Thank you so much, @danielshahaf; sorry for being so inept at this and taking up your time with something so trivial and non-bug related. 😅

**Comment by danielshahaf** [comment]
> That was it, indeed. I set `core.autocrlf` to `input` instead of `true`, 
> re-did the `git clone` of the repo, and now `0a` is the only response to 
> your debugging line. Now the plugin is loading up correctly. Thank you 
> so much, @danielshahaf; sorry for being so inept at this and taking up 
> your time with something so trivial and non-bug related. 😅

You're welcome!  And look into putting your dotfiles in version control
(or configuration management) to make such issues less likely to occur. :)
